tensorflow/core/kernels/bias_op_gpu.cu.cc

```diff
@@ -173,10 +173,10 @@ __global__ void BiasGradNCHW_SharedAtomics(const T* output_backprop,
   // Accumulate the results in the shared memory into the first element.
   // No syncthreads is needed since this is only in the same warp.
   int32 thread_index = threadIdx.x;
-  if (thread_index < 16) s_data[thread_index] += s_data[thread_index + 16];
-  if (thread_index < 8) s_data[thread_index] += s_data[thread_index + 8];
-  if (thread_index < 4) s_data[thread_index] += s_data[thread_index + 4];
-  if (thread_index < 2) s_data[thread_index] += s_data[thread_index + 2];
+  if (thread_index < 16) s_data[thread_index] += s_data[thread_index + 16]; __syncwarp(0xFFFF);
```

Comment:
Please put the __syncwarp calls on a separate line.  As written it looks like the if statement applies to both statements, but it doesn't.